💎(site) Add New Design to Technical Page Hero

Vanss472 opened the pull request:
Added new design to technical page hero.

Hero notes: 
1. Since the image should appear on the right side of the screen, I have removed the position variable from markup.
2. I used the section modifier to target each collection and apply the respective hero color
3. I'll need the final path for the SVGs images `ix://landings/technical/`


- Resolves #338

Commit:
add new styles for hero

Commit:
Merge branch 'redesign/2022' of https://github.com/chromeos/chromeos.dev into feature/technical-page-hero

Commit:
init graphic work on hero

Commit:
add new hero images to all tech-detail index pages

Commit:
clean up and add section modifier as class

Commit:
clean up image class

Commit:
change var to use themeing colors instead

Snugug committed:
Merge branch 'redesign/2022' into feature/technical-page-hero

Commit:
Merge branch 'redesign/2022' of https://github.com/chromeos/chromeos.dev into feature/technical-page-hero

Commit:
fix bg pattern image and clean up path for bg pattern

Commit:
Merge branch 'feature/technical-page-hero' of https://github.com/chromeos/chromeos.dev into feature/technical-page-hero

Commit:
update bg pattern path

Commit:
add todo comment

Commit:
Merge branch 'redesign/2022' of https://github.com/chromeos/chromeos.dev into feature/technical-page-hero

Commit:
update hero image path

Commit:
fix error

Commit:
add recommended section bg and cleanup

Snugug commented on site/sass/layouts/_tech-detail.scss:
```diff
@@ -36,36 +39,54 @@
-      @include mq($bkpt-tech-details--landing) {
-        padding-bottom: 4rem;
+      @include mq($bkpt-tech-details--3-cols) {
+        height: 40vh;
```
I think this needs to be `max-height` or the background image starts to break

```suggestion
        max-height: 40vh;
```

Snugug commented on site/en/linux/index.md:
```diff
@@ -2,8 +2,7 @@
 title: Overview
```
```suggestion
title: Linux on Chrome OS
```

Like the change you made to Productivity. Let's do the same for Linux

Snugug commented on site/sass/layouts/_tech-detail.scss:
```diff
@@ -36,36 +39,54 @@
+        display: flex;
+        justify-content: center;
+        padding-bottom: 0;
+        padding-right: 0;
```
I think it needs to retain this padding or lest the image and text get too close

Vanss472 replied:
Fixed.

Snugug commented on site/_layouts/tech-detail.njk:
```diff
@@ -2,43 +2,38 @@
 </article>
-{% from 'back-to-top.njk' import backToTop %} {{ backToTop() }}
+{% if landingPage == false %}
+<section class="tech-detail__recommended wrapper wrapper--padding wrapper--padded wrapper--full-bleed wrapper__contained wrapper__background">{{itemGrid(microcopy.recommended, title | related(locale.code, page.url))}}</section>
```
```suggestion
<aside class="tech-detail__recommended wrapper wrapper--padding wrapper--padded wrapper--full-bleed wrapper__contained wrapper__background">{{itemGrid(microcopy.recommended, title | related(locale.code, page.url))}}</aside>
```
This probably should have been an `aside` to begin with, but def should be one now

Snugug reviewed (changes requested):
Thanks! This is all looking great! I've got a couple of comments in-line, but the incoming need to swap images for single and double columns may mean the hero image stuff needs to change to the theme stuff we talked about. TBD.

In addition to the inline comments, I've got one more:
* Below about 470px the background image starts to become detached from the bottom right corner

Thanks!

Commit:
Update site/sass/layouts/_tech-detail.scss

Co-authored-by: Sam Richard <snugug@google.com>

Commit:
Update site/en/linux/index.md

Co-authored-by: Sam Richard <snugug@google.com>

Commit:
Update site/_layouts/tech-detail.njk

Co-authored-by: Sam Richard <snugug@google.com>

Commit:
swap images for single and double columns

Commit:
update hero image path

Snugug committed:
Fix image paths

Snugug committed:
Prevent images from getting cut off

Putting the image on a grid and having it be contained was causing it to get cut off. Positioning it like this resolves that problem

Also only apply the background color to the hero class, not everything with a hero in its class name

Snugug commented:
CI is stalled. Going to merge.
